frontend/javascript/components/docupload/attachment-actions.vue

```diff
@@ -0,0 +1,278 @@
+  : null
+)
+
+const canDelete = computed(() => attachment.can_delete && !attachment.approving && !attachment.document)
+
+const canRedact = computed(() => attachment.can_redact)
+
+const isMakingResult = ref(false)
+
+const canMakeResult = computed(() => {
+  if (!config.settings.can_make_document) {
+    return false
+  }
+  return !attachment.document && // not yet made result
+    attachment.approved &&
+    attachment.is_pdf &&
+    !attachment.redacted &&
+    !attachment.converted
+})
+
+const makeResult = () => {
+  createDocument(attachment)
+    .finally(() => {
+      isMakingResult.value = false
+      emit('actionDone')
+    })
+}
+
+const dropdownHasItems = computed(() => canRedact.value || unredacted.value || unconverted.value)
+
+const pdfRedaction = ref()
+
+const pdfRedactionAtt = ref(null)
+
+const pdfRedactionUploaded = () => {
+  // handle v-if'ed modal
+  pdfRedactionModal.value.hide()
+  pdfRedactionAtt.value = null
+  refreshAttachments()
+  emit('actionDone')
+}
+
+const pdfRedactionModal = ref()
+
+const redactClick = (evt, att) => {
+  evt.preventDefault()
+  // handle v-if'ed modal
+  pdfRedactionAtt.value = att
+  nextTick().then(() => pdfRedactionModal.value.show())
+}
+
+const deleteClick = () => {
+  if (!window.confirm(i18n.value.confirmDelete)) return
+  emit('actionDelete')
+  deleteAttachment(attachment)
+  // can't .then-emit actionDone here because the component will be optimistically gone by then
+  // and the modal's backdrop will remain in dom
+}
+
+const makeRelevantClick = () => {
+  makeRelevant(attachment)
+  emit('actionDone')
+}
+
+const approveClick = () => {
+  if (!window.confirm(i18n.value.confirmApproveUnredacted)) return
+  approveAttachment(attachment)
+    .then(() => emit('actionDone'))
+}
+
+</script>
+
+<template>
+  <button v-if="canMakeResult" @click="makeResult" type="button" class="btn btn-sm btn-link text-start"
+    :class="{ disabled: isMakingResult }">
+    <i class="fa fa-certificate"></i>
+    {{ i18n.markResult }}
+  </button>
+  <a
+    v-if="!dropdown && canRedact && !attachment.is_redacted"
+    :href="getRedactUrl(attachment)"
+    class="btn btn-sm btn-link text-start"
+    @click="redactClick($event, attachment)"
+    >
+    <i class="fa fa-square"></i>
+    {{ i18n.redact }}
+  </a>
+  <!-- TODO: this is not really schwärzung bearbeiten, but instead ungeschwärztes original noch mal schwärzen -->
+  <a
+    v-if="!dropdown && unredacted && unredacted.can_redact"
+    :href="getRedactUrl(unredacted)"
+    class="btn btn-sm btn-link text-start"
+    @click="redactClick($event, unredacted)"
+    >
+    <i class="fa fa-pencil-square"></i>
+    {{ i18n.editRedaction }}
+  </a>
+  <button v-if="attachment.is_irrelevant" @click="makeRelevantClick" type="button" class="btn btn-sm btn-link text-start">
+    <i class="fa fa-exclamation-circle"></i>
+    {{ i18n.markNotIrrelevant }}
+  </button>
+  <a class="btn btn-sm btn-link text-start" :href="attachment.file_url" download>
```

Comment:
Right now, the default file names when downloading are the file hashes. Better to provide the name:

```suggestion
  <a class="btn btn-sm btn-link text-start" :href="attachment.file_url" :download="attachment.name">
```

(more of this below)

Reply:
when I test this locally, both FF and Chromium ignore this, and prefer the filename from the endpoint's header:

`http://localhost:8000/files/foi/900123/brief_geschwaerzt.pdf?token=f0ob4r` ⇒  
`Content-Disposition: inline; filename="d3adb33f….pdf"`

Reply:
Oh, didn't realize we set that using headers.

Reply:
attempted in 54f574ad3169c401f43329d63a24d005f925f83b if you want to give it a try